src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/AddKeepAlivesStep.cs

```diff
@@ -10,17 +10,30 @@
 using Mono.Linker.Steps;
 
 using Mono.Cecil.Cil;
+#if NET5_LINKER
+using Microsoft.Android.Sdk.ILLink;
+#endif  // NET5_LINKER
 
 namespace MonoDroid.Tuner
 {
 	public class AddKeepAlivesStep : BaseStep
 	{
+
+#if NET5_LINKER
+		LinkContextMetadataResolver cache;
```

Comment:
If we change `cache` to be an `IMetadataResolver`, that will reduce what the `#if NET5_LINKER` needs to "protect"/differentiate.  `AddKeepAliveStep()` could likewise us an `IMetadataResolver cache` field.